tests/test_box_mesh.py

```diff
@@ -232,3 +234,86 @@ def test_given_box_box_mesh_boundary_elements_must_find_boundary_surface_element
     assert boundary.n_cells == expected_number_of_cells and all(
         bool_check_triangle_on_boundary_list
     )
+
+
+@pytest.mark.parametrize("k_neighbours", [1, 2, 3, 4])
+def test_closest_points_on_boundaries_periodic_mesh_must_have_only_neighbour(
+    k_neighbours: int,
+) -> None:
+    box_mesh_to_test = BoxMesh(_box_mesh_points(), _box_mesh_elements())
+    closest_pts = box_mesh_to_test.closest_points_on_boundaries(
+        k_neighbours=k_neighbours
+    )
+
+    number_of_closest_neighbours_to_test: List[int] = []
+    number_closest_neighbours_truth_1_on_each_face_and_edge = [1] * 12
+    for key, list_of_neighbor_for_all_nodes in closest_pts.items():
+        list_of_neighbor_for_all_nodes_index = list_of_neighbor_for_all_nodes[0]
+
+        number_of_closest_neighbours_to_test.extend(
+            len(list_of_neighbor_for_each_nodes)
+            for list_of_neighbor_for_each_nodes in list_of_neighbor_for_all_nodes_index
+        )
+
+    assert (
+        number_of_closest_neighbours_to_test
+        == number_closest_neighbours_truth_1_on_each_face_and_edge
+    )
+
+
+@pytest.mark.parametrize("k_neighbours", [1, 2, 3, 4])
+def test_closest_points_on_perturbated_mesh_boundaries_must_have_good_number_of_neighbours(
+    k_neighbours: int,
+) -> None:
+    box_mesh_to_test = BoxMesh(_box_mesh_points(), _box_mesh_elements())
+    perturbation_factor = 0.01
+
+    xyz_min_values_of_rve = np.min(box_mesh_to_test.rve.min_point)
+    xyz_max_values_of_rve = np.max(box_mesh_to_test.rve.max_point)
+
+    # perturb faces and edges with non-zero perturbation so that multiple neighbours must be found
+    for key, node_index_in_faces in box_mesh_to_test.faces.items():
+        box_mesh_to_test.nodes_coords[node_index_in_faces[0]] += perturbation_factor * (
+            1 - random.uniform(0, 1)
+        )
+
+    for key, node_index_in_edges in box_mesh_to_test.edges.items():

```

Comment:
```suggestion
    for _, node_index_in_edges in box_mesh_to_test.edges.items():

```

As the variable name "key" is not used in this loop you can ignored it using "_"